multicluster/controllers/multicluster/stale_controller.go

```diff
@@ -416,9 +405,15 @@ func (c *StaleResCleanupController) Run(stopCh <-chan struct{}) {
 	defer klog.InfoS("Shutting down StaleResCleanupController")
 
 	ctx, _ := wait.ContextForChannel(stopCh)
-	if err := c.RunOnce(ctx); err != nil {
-		c.Enqueue()
-		go wait.UntilWithContext(ctx, c.runWorker, time.Second)
+
+	switch c.clusterRole {
+	case LeaderCluster:
+		go wait.UntilWithContext(ctx, c.cleanupMemberClusterAnnounces, 12*time.Hour)
```

Comment:
Do not hardcode 12 hours. Maybe use "memberClusterAnnounceStaleTime / 2"

Reply:
`cleanupMemberClusterAnnounces` only runs once after MC Controller starts? Maybe add a comment about this in line 52.

Reply:
`c.cleanupMemberClusterAnnounces` will run every 12 hours in the leader cluster, but the stale controller will run only once in the member cluster if it succeeds. I will update the comment for the member cluster.